api/v1alpha1/database_webhook.go

```diff
@@ -0,0 +1,164 @@
+
+var _ webhook.Defaulter = &Database{}
+
+func defaulterDatabaseCreate_NDBSpec(r *Database) {
+	if !r.Spec.NDB.SkipCertificateVerification {
+		r.Spec.NDB.SkipCertificateVerification = false
+	}
+}
+
+func defaulterDatabaseCreate_InstanceSpec(r *Database) {
+
+	if r.Spec.Instance.DatabaseInstanceName == "" {
+		r.Spec.Instance.DatabaseInstanceName = common.DefaultDatabaseInstanceName
+	}
+
+	if len(r.Spec.Instance.DatabaseNames) == 0 {
+		r.Spec.Instance.DatabaseNames = common.DefaultDatabaseNames
+	}
+
+	if r.Spec.Instance.Size == 0 {
+		r.Spec.Instance.Size = 10
+	}
+
+	if r.Spec.Instance.TimeZone == "" {
+		r.Spec.Instance.TimeZone = "UTC"
+	}
+
+}
+
+// Default implements webhook.Defaulter so a webhook will be registered for the type
+func (r *Database) Default() {
+
+	databaselog.Info("Entering Default...")
+
+	databaselog.Info("default", "name", r.Name)
+	defaulterDatabaseCreate_NDBSpec(r)
+	defaulterDatabaseCreate_InstanceSpec(r)
+
+	databaselog.Info("Exiting Default...")
+}
+
+// TODO(user): change verbs to "verbs=create;update;delete" if you want to enable deletion validation.
+//+kubebuilder:webhook:path=/validate-ndb-nutanix-com-v1alpha1-database,mutating=false,failurePolicy=fail,sideEffects=None,groups=ndb.nutanix.com,resources=databases,verbs=create;update,versions=v1alpha1,name=vdatabase.kb.io,admissionReviewVersions=v1
+
+var _ webhook.Validator = &Database{}
+
+func validateDatabaseCreate_NDBSpec(r *Database, allErrs field.ErrorList, ndbPath *field.Path) field.ErrorList {
+	databaselog.Info("Entering validateDatabaseCreate_NDBSpec...")
+	if r.Spec.NDB == (NDB{}) {
+		allErrs = append(allErrs, field.Invalid(ndbPath, r.Spec.NDB, "NDB spec must not be empty"))
+	}
+
+	if err := util.ValidateUUID(r.Spec.NDB.ClusterId); err != nil {
+		allErrs = append(allErrs, field.Invalid(ndbPath.Child("clusterId"), r.Spec.NDB.ClusterId, "ClusterId field must be a valid UUID"))
+	}
+
+	if r.Spec.NDB.CredentialSecret == "" {
+		allErrs = append(allErrs, field.Invalid(ndbPath.Child("credentialSecret"), r.Spec.NDB.CredentialSecret, "CredentialSecret must not be empty"))
+	}
+
+	if err := util.ValidateURL(r.Spec.NDB.Server); err != nil {
+		allErrs = append(allErrs, field.Invalid(ndbPath.Child("server"), r.Spec.NDB.Server, "Server must be a valid URL"))
+	}
+
+	databaselog.Info("Exiting validateDatabaseCreate_NDBSpec...")
+	return allErrs
+}
+
+func validateDatabaseCreate_NewDBSpec(r *Database, allErrs field.ErrorList, instancePath *field.Path) field.ErrorList {
+	databaselog.Info("Entering validateDatabaseCreate_NewDBSpec...")
+
+	if r.Spec.Instance.DatabaseInstanceName == "" {
+		allErrs = append(allErrs, field.Invalid(instancePath.Child("databaseInstanceName"), r.Spec.Instance.DatabaseInstanceName, "Database Instance Name must not be empty"))
+	}
```

Comment:
Can we remove this validation since the defaulter will make sure that this value is never empty?

Reply:
This can be discussed as a part of the additional line item that I have added in the operator task sheet. The default values or mandatory fields need some discussion so I will close it here.